driver-core/src/main/com/mongodb/client/model/geojson/Geometry.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.geojson;
+
+import com.mongodb.client.model.geojson.codecs.GeoJsonCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistries;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.json.JsonWriter;
+import org.bson.json.JsonWriterSettings;
+
+import java.io.StringWriter;
+
+/**
+ * An abstract class for representations of GeoJSON geometry objects.
+ *
+ * @since 3.1
+ */
+public abstract class Geometry {
```

Comment:
Two comments:

1. Do you not want to prefix your GeoJson classes with GeoJson to avoid ambiguity or name collissions?
2. Why does this not have GeoJsonObject as a base class? I realize that since you chose not to implement GeoJsonFeature (yet) that GeoJsonGeometry would be the only GeoJsonObject subclass so far, but it seems like the right thing to do is to have a GeoJsonObject class.